plugins/sign-on/openid-connect-generic-client/includes/openid-connect-generic-client-wrapper.php

```diff
@@ -424,7 +424,8 @@ function authentication_request_callback() {
 
 		if ( ! $user ) {
 			if ( $this->settings->create_if_does_not_exist ) {
-				$user = $this->create_new_user( $subject_identity, $user_claim );
+				$user = $this->get_common_login($user_claim);
```

Comment:
Could consider changing the condition here from "create_if_does_not_exist" to simply "does not exist and authenticated". Since we're not really going to create a new user every time the "create if does not exist" might not be so intuitive anymore. Not a big deal though - just a suggestion

Reply:
I get your point but since we handle the authentication we know the user is not logged in and since we are inside of !$user check condition we know the user does not exist. I will remove the create_if_does_not_exist check for now.

Reply:
Will as well remove the check for is_wp_error since we already do that inside of the get_common_login function.

Reply:
create_if_does_not_exist could be used to create users, instead of using a common/shared user (as required for frabegynnelsen). The difference would still be that we don't populate those created users with a lot of person data (like personal email address), name etc.)

Reply:
Do we actually want to release that option to everyone so we still allow them to create a WP user for every BCC member (although with less data)? Or just create a custom version of the plugin for the frabegynnelsen project?
Another thing we could do is make the **create_if_does_not_exist** setting readonly, and set it to false.

Reply:
As much as possible, we should have one version of the plugin. But I like the idea of setting the option to false / readonly if it can easily be overridden in code when necessary. (Another option would be to have it false by default, but still allow admins to override the setting.)

Reply:
Alright. I've added the possibility to still create users in WP if you override the value in code, while the setting is set to false and readonly.